tsconfig.json

```diff
@@ -6,6 +6,7 @@
     "moduleResolution": "node",
     "esModuleInterop": true,
     "resolveJsonModule": true,
+    "downlevelIteration": true,
```

Comment:
Interesting, whats the reasoning for adding this?

Reply:
I used it to suppress the Typescript warning when trying to iterate through KVs in a map for the compareMaps function: https://github.com/Theodo-UK/sustainability-calculator/commit/df559e1ec389fc41f57f6450543f19c87587c53b

![image](https://github.com/Theodo-UK/sustainability-calculator/assets/57725347/c8a3761f-d3d5-4740-8a64-aa26b97c8e66)

Might not be needed if the compareMaps function changes?

Reply:
https://www.typescriptlang.org/tsconfig#downlevelIteration

According to the typescript documentation, it is converting typescript to handle an older version of JS. Which is kinda interesting to find out why it is needed?

Reply:
Yes, because I think the syntax `for (let [key, val] of map1) {...}` is not supported by older versions of JS

Alternatively I could use forEach and remove the downleveliteration flag:

```typescript
export function compareMaps(map1: Map<any, any>, map2: Map<any, any>) {
    let result = true;
    if (map1.size !== map2.size) {
        return false;
    }
    map1.forEach((val, key) => {
        console.log(key, val)
        let testVal = map2.get(key);
        console.log(key, testVal)
        // in cases of an undefined value, make sure the key
        // actually exists on the object so there are no false positives
        if (testVal !== val || (testVal === undefined && !map2.has(key))) {
            result = false;
            return 
        }
    });
    return result;
}
```